src/analysis/tuneParam/tuneParam.m

```diff
@@ -0,0 +1,158 @@
+function optimParam = tuneParam(LPProblem,contFunctName,timelimit,nrepeat,printLevel)
+% Optimizes cplex parameters to make model resolution faster.
+% Particularly interetsing for large-scale MILP models and repeated runs of
+% optimisation.
+% While, it won't optimize memory space nor model constraints for numerical
+% infeasibilities, tuneParam will provide the optimal set of solver
+% parameters for feasbile models. It requires IBM ILOG cplex (for now).
+%
+% USAGE
+%
+%    cpxControl = CPLEXParamSet('ILOGcomplex');
+%    load ecoli_core_model;
+%    optimalParameters = tuneParam(model,cpxControl,1000,1000,0);
+%
+% INPUT:
+%         LPProblem:     MILP as COBRA model structure
+%         contFunctName: Parameters structure containing the name and value.
+%                        A set of routine parameters will be added by the solver
+%                        but won't be reported.
+%         timelimit:     default is 10000 second
+%         nrepeat:       number of row/column permutation of the original
+%                        problem, reports robust results.
+%                        sets the CPX_PARAM_TUNINGREPEAT parameter
+%                        High values of nrepeat would require consequent
+%                        memory and swap.
+%         printLevel:    0/1/2/3
+%
+% OUTPUT:
+%         optimParam: structure of optimal parameter values directly usable as
+%                     contFunctName argument in solveCobraLP function
+% .. Author: Marouen Ben Guebila 24/07/2017
+
+if ~changeCobraSolver('ibm_cplex')
+    fprintf('This function requires IBM ILOG CPLEX');
+end
+if exist('timelimit','var')
+    contFunctName.tune.timelimit = timelimit;
+end
+if exist('nrepeat','var')
+    contFunctName.tune.repeat = nrepeat;
+end
+if exist('printLevel','var')
+    contFunctName.tune.display = printLevel;
+end
+interface='ILOGcomplex';
+if ~exist('contFunctName','var')
```

Comment:
Shouldn't this be checked first? 
Or is the intention, that a custom function + specific parameters is not allowed?
Currently, if contFuncName is a callable function and additional arguments are supplied, this throws an error with an odd error message. If you flip the initialisation, the individual parameters would override whatever comes from the function. 
I would actually check, that contFuncName is either empty or a char, as otherwise error messages would be odd.